src/scripts/main.js

```diff
@@ -1 +1,83 @@
 'use strict';
+
+function createFirstPromise() {
+  return new Promise((resolve, reject) => {
+    let isClicked = false;
+
+    document.addEventListener('click', () => {
+      isClicked = true;
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      if (!isClicked) {
+        reject(new Error('First promise was rejected'));
+      }
+    }, 3000);
+  });
+}
+
+function createSecondPromise() {
+  return new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('Second promise was resolved');
+    });
+
+    document.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+    });
```

Comment:
Similar to the click event, adding multiple event listeners for 'contextmenu' can lead to unexpected behavior. Consider removing the listener after it resolves the promise to prevent multiple resolutions.